core/citrus-api/src/main/java/com/consol/citrus/spi/ResourcePathTypeResolver.java

```diff
@@ -130,6 +141,32 @@ public <T> Map<String, T> resolveAll(String resourcePath, String property, Strin
         return resources;
     }
 
+    private static List<Resource> resolveAllFromJar(String path) {
+        URL root = ResourcePathTypeResolver.class.getProtectionDomain().getCodeSource().getLocation();
+        List<Resource> result = new ArrayList<>();
+        String rootAsString = root.toString();
+        if (rootAsString.endsWith(".jar") && !rootAsString.matches(".*" + File.separator + "citrus-api-\\d+\\.\\d+\\.\\d+(-.*)?\\.jar")) {
+            try (ZipInputStream in = new ZipInputStream(root.openStream())) {
+                while (true) {
```

Comment:
thanks very much for the contribution (and the detailed description in the issue)! there is just one thing I would like to discuss. the `while(true)` part looks ugly in my opinion. what about this?

```java
ZipEntry entry;
while ((entry = in.getNextEntry()) != null) { // or !Objects#isNull if you prefer that
    if (entry.getName().startsWith(path)) {
        Optional.of(ResourcePathTypeResolver.class)
            .map(Class::getClassLoader)
            .map(classLoader -> classLoader.getResource(entry.getName()))
            .map(UrlResource::new)
            .ifPresent(result::add);
    }
}
```

Reply:
This won't work since the usage of `entry` in the lambda needs it to be final or effectively-final. We would need to rewrite the `Optional`-part.

Reply:
that is true, I forgot that. you could use an [atomic reference](https://docs.oracle.com/javase/8/docs/api/java/util/concurrent/atomic/AtomicReference.html) for this purpose, which you can `set` and `get`. but that again might be too much trouble for what it is 🙁 thoughts @christophd?

Reply:
@bbortt I refactored the code a little (added caching for the zip entry so the zip is not read multiple times). This moved the problematic code in a non-lambda part and I applied your change. Care to take another look?